### PR TITLE
Correct handle the value return from mgmt socket read and write (#6220)

### PR DESCRIPTION
**Pull request by oknet**

### CODE REVIEW

File: mgmt/utils/MgmtUtils.cc [hunk 1]
```diff
@@ -70,11 +70,7 @@ mgmt_readline(int soc, char *buf, int maxlen)
         break;
       }
     } else if (rc == 0) {
-      if (n == 1) { /* EOF */
```
Comment (oknet) [inline mgmt/utils/MgmtUtils.cc]: This also fix a bug was introduced by PR #1014 / TS-4861.

**Comment by oknet** [comment]
[approve ci autest]

File: mgmt/utils/MgmtUtils.cc [hunk 2]
```diff
@@ -112,7 +108,7 @@ mgmt_writeline(int soc, const char *data, int nbytes)
   while (nleft > 0) {
     int nwritten = write_socket(soc, tmp, nleft);
     if (nwritten <= 0) { /* Error or nothing written */
-      return nwritten;
+      return nleft;
```
Comment (oknet) [inline mgmt/utils/MgmtUtils.cc]: Returns: num bytes not written

File: mgmt/utils/MgmtUtils.cc [hunk 3]
```diff
@@ -105,24 +101,39 @@ mgmt_readline(int soc, char *buf, int maxlen)
...
+      continue;
+    } else if (n < 0) { // Error
+      if (mgmt_transient_error()) {
+        mgmt_sleep_msec(1);
+        continue;
+      }
+
       return n;
-    } else {
-      return (nbytes - nleft);
```
Comment (oknet) [inline mgmt/utils/MgmtUtils.cc]: Returns:    num bytes not written

**Comment by vmamidi** [comment]
What is the problem that we are trying to solve here?

**Comment by zwoop** [comment]
[approve ci].

**Comment by oknet** [comment]
@vmamidi

1. Fix a bug was introduced by PR #1014 / TS-4861 (v7.0.0).
2. Other minor bugs found through code review.

**Pull request merged**